aea/protocols/generator.py

```diff
@@ -1807,7 +2646,8 @@ def generate(self) -> None:
         self._generate_file(INIT_FILE_NAME, self._init_str())
         self._generate_file(PROTOCOL_YAML_FILE_NAME, self._protocol_yaml_str())
         self._generate_file(MESSAGE_DOT_PY_FILE_NAME, self._message_class_str())
-        if len(self._all_custom_types) > 0:
+        self._generate_file(DIALOGUE_DOT_PY_FILE_NAME, self._dialogue_class_str())
```

Comment:
Dialogue file should only be generated if specified?

Reply:
Done